app/liquid/liquid_renderer.rb

```diff
@@ -43,6 +44,15 @@ def images
 
   private
 
+  def set_locale
+    begin
+      if @page.language.present? && @page.language.code.present?
+        I18n.locale = @page.language.code
+      end
+    rescue I18n::InvalidLocale
```

Comment:
ahh yeah. this just makes it so it only sets the locale if it's one of the ones explicitly set in config. I'll add a comment.